spec/board_spec.rb

```diff
@@ -0,0 +1,65 @@
+require 'board'
+
+describe Board do
+  subject(:board) { described_class.new([[1, 2, 3], [4, 5, 6], [7, 8, 9]]) }
+  subject(:board2) { described_class.new([%i[X O X], %i[X X O], %i[O X O]]) }
+
+  context 'when initialized' do
+    it 'accepts a grid' do
+      expect { Board.new([[1, 2, 3], [4, 5, 6], [7, 8, 9]]) }.not_to raise_error
+    end
+
+    it 'has a default grid with 3 rows' do
+      expect(board.grid.size).to eq 3
```

Comment:
Test behaviour, not state: Objects are collections of behaviour that happen to need state, not collections of state.